### ISSUE TITLE
Fail materialized future in FileIO.toPath API when IO operation fails

### ISSUE THREAD

**Issue by jrudolph** [akka/akka#23951]
Right now it materializes to a `Future[IOResult]`. Contrary to expectations, it will not return a failed future when the operation fails but it will return an `IOResult` with a failed status field. It should also be configurable what should happen with files that haven't been completely written.

**Comment by ennru** [comment]
To get IO errors in there might be acceptable, but even errors from upstream disappear into `IOResult`.
```
val future = Source
  .failed(new RuntimeException("error"))
  .runWith(FileIO.toPath(targetDirectory.resolve("tmp")))
```
The future succeeds with the error hidden in the `IOResult`.

**Comment by jrudolph** [comment]
This is still broken.

**Comment by jrudolph** [comment]
https://github.com/akka/akka-http/issues/1488 is a direct consequence of the broken behavior.

**Comment by johanandren** [comment]
I think this was fixed by #26811 (although I see that the last comment from @jrudolph was after that, hmmm)
